src/python/pants/backend/helm/util_rules/sources.py

```diff
@@ -37,8 +36,13 @@ def debug_hint(self) -> str | None:
         return self.source.address.spec
 
 
+@dataclass(frozen=True)
+class HelmChartSourceRoot:
```

Comment:
Optional: Maybe name this `HelmChartRoot` (and `HelmChartRootRequest`) to make it more clear to the reader that these are not source roots as that term is usually used.

Reply:
I'll do it, I do like all these suggestions!